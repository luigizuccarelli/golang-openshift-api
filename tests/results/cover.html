
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>handlers: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go (74.2%)</option>
				
				<option value="file1">github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go (92.9%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package handlers

import (
        "bytes"
        "fmt"
        "io"
        "io/ioutil"
        "net/http"
        "os"
        "os/exec"
        "regexp"
        "strconv"

        "github.com/luigizuccarelli/golang-openshift-api/pkg/connectors"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

func APICall(api string, token string, conn connectors.Client) ([]byte, error) <span class="cov8" title="1">{
        req, _ := http.NewRequest("GET", api, nil)
        req.Header.Set("Authorization", "Bearer "+token)
        req.Header.Set("Content-Type", "application/json")
        resp, err := conn.Do(req)
        if err != nil </span><span class="cov0" title="0">{
                conn.Error(fmt.Sprintf("Function APICall http request %v", err))
                return []byte(""), err
        }</span>

        <span class="cov8" title="1">defer resp.Body.Close()
        body, e := ioutil.ReadAll(resp.Body)
        if e != nil </span><span class="cov0" title="0">{
                conn.Error(fmt.Sprintf("Function APICall %v", e))
                return []byte(""), err
        }</span>
        <span class="cov8" title="1">conn.Debug(fmt.Sprintf("Function APICall response %s", string(body)))
        return body, nil</span>
}

func ExecJQFilter(input []byte, filter string, conn connectors.Client) (string, error) <span class="cov8" title="1">{
        // clean up first
        err := ioutil.WriteFile("/tmp/test.json", input, 0755)
        if err != nil </span><span class="cov0" title="0">{
                conn.Error("cleanup up tmp file %v", err)
        }</span>
        <span class="cov8" title="1">result, err := ExecOS(os.Getenv("HOME_DIR"), "jq", []string{filter, "/tmp/test.json"}, false)
        conn.Debug("Filter response %s", result)
        if err != nil </span><span class="cov0" title="0">{
                conn.Error("filter error %v", err)
                return "", err
        }</span>
        <span class="cov8" title="1">return result, nil</span>
}

func ExecRegex(input []byte, regex string, conn connectors.Client) [][]byte <span class="cov8" title="1">{
        reg := regexp.MustCompile(regex)
        res := reg.FindAll(input, -1)
        for x, _ := range res </span><span class="cov8" title="1">{
                conn.Info("Result %v", string(res[x]))
        }</span>
        <span class="cov8" title="1">return res</span>
}

func ValidateResults(input [][]byte, occurence string, operand string) string <span class="cov8" title="1">{
        var result string = ""
        occ, _ := strconv.Atoi(occurence)
        switch operand </span>{
        case "eq":<span class="cov8" title="1">
                if occ == len(input) </span><span class="cov8" title="1">{
                        result = "PASS"
                }</span> else<span class="cov8" title="1"> {
                        result = "FAIL"
                }</span>
        case "lt":<span class="cov8" title="1">
                if occ &lt; len(input) </span><span class="cov8" title="1">{
                        result = "PASS"
                }</span> else<span class="cov8" title="1"> {
                        result = "FAIL"
                }</span>
        case "gt":<span class="cov8" title="1">
                if occ &gt; len(input) </span><span class="cov8" title="1">{
                        result = "PASS"
                }</span> else<span class="cov8" title="1"> {
                        result = "FAIL"
                }</span>
        case "ne":<span class="cov8" title="1">
                if occ != len(input) </span><span class="cov8" title="1">{
                        result = "PASS"
                }</span> else<span class="cov8" title="1"> {
                        result = "FAIL"
                }</span>
        }
        <span class="cov8" title="1">return result</span>
}

// ExecOS - call to shell
func ExecOS(path string, c string, params []string, trim bool) (string, error) <span class="cov8" title="1">{
        var stdout, stderr bytes.Buffer
        var out string = ""
        cmd := exec.Command(c, params...)
        cmd.Dir = path
        cmd.Stdout = &amp;stdout
        cmd.Stderr = &amp;stderr
        err := cmd.Run()
        outStr, errStr := string(stdout.Bytes()), string(stderr.Bytes())
        if err != nil </span><span class="cov0" title="0">{
                return errStr, err
        }</span>
        <span class="cov8" title="1">if trim </span><span class="cov0" title="0">{
                out = outStr[:len(outStr)-1]
        }</span> else<span class="cov8" title="1"> {
                out = outStr
        }</span>
        <span class="cov8" title="1">return out, nil</span>
}

func PipeOutput(input string) error <span class="cov0" title="0">{
        r, w := io.Pipe()
        go func() </span><span class="cov0" title="0">{
                fmt.Fprint(w, input)
                w.Close()
        }</span>()

        <span class="cov0" title="0">_, err := io.Copy(os.Stdout, r)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>
        <span class="cov0" title="0">return nil</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package validator

import (
        "fmt"
        "os"
        "strconv"
        "strings"

        "github.com/microlib/simple"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *simple.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("Input paramaters -&gt; name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return fmt.Errorf(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span>

                <span class="cov0" title="0">logger.Error(fmt.Sprintf("%s envar is empty please set it", name))</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
func ValidateEnvars(logger *simple.Logger) error <span class="cov8" title="1">{
        items := []string{
                "HOME_DIR,true",
                "NAME,false",
                "VERSION,true",
        }
        for x := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
