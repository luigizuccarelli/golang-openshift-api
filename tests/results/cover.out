mode: set
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:13.59,17.27 4 1
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:25.2,25.12 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:17.27,18.15 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:23.3,23.69 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:18.15,21.4 2 0
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:30.50,37.23 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:42.2,42.12 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:37.23,38.54 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/validator/validate.go:38.54,40.4 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:22.80,27.16 5 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:32.2,34.14 3 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:38.2,39.18 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:27.16,30.3 2 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:34.14,37.3 2 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:42.88,45.16 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:48.2,50.16 3 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:54.2,54.20 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:45.16,47.3 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:50.16,53.3 2 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:57.77,60.24 3 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:63.2,63.12 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:60.24,62.3 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:66.95,68.17 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:107.2,107.15 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:69.12,70.25 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:85.12,87.23 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:92.12,94.23 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:99.12,101.24 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:70.25,71.28 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:76.4,76.17 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:71.28,72.36 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:72.36,74.6 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:77.9,79.25 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:79.25,81.5 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:81.10,83.5 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:87.23,89.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:89.9,91.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:94.23,96.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:96.9,98.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:101.24,103.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:103.9,105.4 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:111.80,120.16 9 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:123.2,123.10 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:128.2,128.17 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:120.16,122.3 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:123.10,125.3 1 0
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:125.8,127.3 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:131.37,133.12 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:138.2,139.16 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:142.2,142.12 1 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:133.12,136.3 2 1
github.com/luigizuccarelli/golang-openshift-api/pkg/handlers/handler.go:139.16,141.3 1 0
